lib/src/trailer.rs

```diff
@@ -0,0 +1,277 @@
+        let descriptions = r#"chore: update itertools to version 0.14.0
+
+Summary: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+tempor incididunt ut labore et dolore magna aliqua.
+
+Change-Id: I1234567890abcdef1234567890abcdef12345678"#;
+
+        let trailers = parse_description_trailers(descriptions);
+
+        // should only have Change-Id
+        assert_eq!(trailers.len(), 1);
+        assert_eq!(trailers[0].key, "Change-Id");
+    }
+
+    #[test]
+    fn test_multiline_trailer() {
+        let description = r#"chore: update itertools to version 0.14.0
+
+key: This is a very long value, with spaces and
+  newlines in it."#;
+
+        let trailers = parse_description_trailers(description);
+
+        // should only have Change-Id
+        assert_eq!(trailers.len(), 1);
+        assert_eq!(trailers[0].key, "key");
+        assert_eq!(
+            trailers[0].value,
+            r"This is a very long value, with spaces and
+  newlines in it."
+        );
+    }
+
+    #[test]
+    fn test_ignore_line_in_trailer() {
+        let description = r#"chore: update itertools to version 0.14.0
+
+Signed-off-by: Random J Developer <random@developer.example.org>
+[lucky@maintainer.example.org: struct foo moved from foo.c to foo.h]
+Signed-off-by: Lucky K Maintainer <lucky@maintainer.example.org>
+"#;
+
+        let trailers = parse_description_trailers(description);
+        assert_eq!(trailers.len(), 2);
+    }
+
+    #[test]
+    fn test_trailers_with_single_line_description() {
+        let description = r#"chore: update itertools to version 0.14.0"#;
+        let trailers = parse_description_trailers(description);
+        assert_eq!(trailers.len(), 0);
+    }
+
+    #[test]
+    fn test_blank_line_after_trailer() {
+        let description = r#"subject
+
+foo: 1
+
+"#;
+        let trailers = parse_description_trailers(description);
+        assert_eq!(trailers.len(), 1);
+    }
+
+    #[test]
+    fn test_blank_line_inbetween() {
+        let description = r#"subject
+
+foo: 1
+
+bar: 2
+"#;
+        let trailers = parse_description_trailers(description);
+        assert_eq!(trailers.len(), 1);
+    }
+
+    #[test]
+    fn test_no_blank_line() {
+        let description = r#"subject: whatever
+foo: 1
+"#;
+        let trailers = parse_description_trailers(description);
+        assert_eq!(trailers.len(), 0);
+    }
+
+    #[test]
+    fn test_whitespace_before_key() {
+        let description = r#"subject
+
+ foo: 1
+"#;
+        let trailers = parse_description_trailers(description);
+        assert_eq!(trailers.len(), 0);
+    }
+
+    #[test]
+    fn test_whitespace_after_key() {
+        let description = r#"subject
+
+foo : 1
+"#;
+        let trailers = parse_description_trailers(description);
+        assert_eq!(trailers.len(), 1);
+        assert_eq!(trailers[0].key, "foo");
+    }
+
+    #[test]
+    fn test_whitespace_around_value() {
+        let description = r#"subject
+
+foo:  1 
```

Comment:
nit: use `\x20` instead of trailing ` `? It's hard to spot.

You can use `indoc! {""}` macro to format multi-line sample nicely.